bundle/deploy/state.go

```diff
@@ -0,0 +1,135 @@
+package deploy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/fileset"
+)
+
+const DeploymentStateFileName = "deployment.json"
+
+type File struct {
+	Path string `json:"path"`
+}
+
+type Filelist []File
+
+type DeploymentState struct {
+	Version   string    `json:"version"`
```

Comment:
Let's use an integer for this one as well. Makes for an easier comparison. We'd also need to check that the deployment state version is not higher than the version supported by the CLI. This could happen if a newer CLI performs a deployment and writes a deployment state with a new schema (e.g. additional fields). If this is the case, we must never interpret the deploy and fail immediately after pulling the state.

Reply:
It also makes sense to include the CLI version in the deployment state. Then we can include it in the error message if there is a deployment state version mismatch (i.e. this bundle was previously deployed by a newer CLI version (vX.Y.Z), which is incompatible with the current CLI version (vX.Y.Z); please upgrade, etc).

Reply:
I don't think we need to do version check now because we don't yet have any logic on what to do on version mismatch. But indeed tracking makes sense

Reply:
Yeah, it's not necessary ATM but It'll be nice to have this as a fast follow-up. By the time you actually need the invalidation semantics it's too late to add it in. 

The logic can be as simple as to have a global integer for current version. If the version pulled is greater, then error asking user to upgrade their CLI.